src/protocol/context/upgrade.rs

```diff
@@ -194,17 +194,17 @@ impl AsRef<str> for Upgrade2DVectors {
 }
 
 #[async_trait]
-impl<'a, C, F, T, M> UpgradeToMalicious<'a, T, Vec<M>> for UpgradeContext<'a, C, F, NoRecord>
+impl<'a, C, F, I, M> UpgradeToMalicious<'a, I, Vec<M>> for UpgradeContext<'a, C, F, NoRecord>
 where
     C: UpgradedContext<F>,
     F: ExtendableField,
-    T: IntoIterator + Send + 'static,
-    T::IntoIter: ExactSizeIterator + Send,
-    T::Item: Send + 'static,
+    I: IntoIterator + Send + 'static,
+    I::IntoIter: ExactSizeIterator + Send,
```

Comment:
`IntoIterator` is a trait that defines shared behavior (convert something into an iterator). `IntoIter` is an associated type for `IntoIterator` trait. 

```rust
pub trait IntoIterator {
    /// The type of the elements being iterated over.
    type Item;

    /// Which kind of iterator are we turning this into?
    type IntoIter: Iterator<Item = Self::Item>;

    /// Creates an iterator from a value.
    fn into_iter(self) -> Self::IntoIter;
}
```

It basically says that `into_iter` must return *some* iterator, but it does not care about *the specific type* of that iterator, because `Iterator` is a trait as well.

In this case, `UpgradeToMalicious::upgrade` method takes `I` which must be convertible into `ExactSizeIterator`